server/include/bge/World.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "Entity.h"
+#include "System.h"
+#include "ComponentManager.h"
+#include "GameConstants.h"
+#include <set>
+#include <iostream>
+
+namespace bge {
+
+    class World {
+        public:
+            void init();
+            Entity createEntity();
+
+            template<typename ComponentType>
+            void addComponent(Entity e, ComponentType c);
+
+            template<typename ComponentType>
+            void deleteComponent(Entity e, ComponentType c);
+
+            void movePlayer(unsigned int player, float x, float y, float z);
+
+            void printDebug();
+
+        private:
+            std::vector<System> systems;
+            std::set<Entity> entities;
+            int currMaxEntityId;
+
+            ComponentManager<PositionComponent> positionCM;
+
+            Entity players[NUM_MOVEMENT_ENTITIES];
```

Comment:
If I'm understanding correctly based on how it's used, NUM_MOVEMENT_ENTITIES seems to specifically refer to the number of players. But I think in the future we will probably have entities that move besides players (e.g. depending on how we handle bullets). Maybe rename this to something more distinct like NUM_PLAYERS?

Reply:
I agree, I think this was just a temporary name Alan came up with.